### ISSUE TITLE
"Invalid RFC-7231 date-time value" when performing GetObjectCommand on S3 Client with Credentials 

### ISSUE THREAD

**Issue by JJSphar** [aws/aws-sdk-js-v3#2977]
## Describe the bug
I get an "Invalid RFC-7231 date-time value" error, when sending a `GetObjectCommand` through an `S3Client`, after having set a `Provider<Credentials>` for the `credentials` parameter.

This only occurs when deployed to Amplify, not when running locally. The error seems to occur when trying to validate the credentials expiration date. 

The error is being thrown from this line: [https://github.com/aws/aws-sdk-js-v3/blame/main/packages/smithy-client/src/date-utils.ts#L127](https://github.com/aws/aws-sdk-js-v3/blame/main/packages/smithy-client/src/date-utils.ts#L127)

## Your environment
React SPA, deployed through Amplify using Github integration.

#### SDK version number
` "@aws-sdk/client-s3": "~3.30.0",`

#### Is the issue in the browser/Node.js/ReactNative?
Browser

#### Details of the browser/Node.js/ReactNative version
Chrome Version 95.0.4638.69 (Official Build) (64-bit)
node -v v14.16.0

## Steps to reproduce
``` 
const client = new S3Client({region: "us-west-1", credentials: getCredentials})
const command = new GetObjectCommand({
    Bucket: env.XXX,
    Key: keyName,
    ResponseCacheControl: "no-cache"
})

const data = await client.send(command)

```

#### Observed behavior
Date-time parsing throws an error for what should be a valid date-time. This only occurs for when hosted on Amplify, when running locally I cannot repro. 

## Additional context
I have a strong hunch this is related to #2866 - in that Issue, a similar date-time parsing error is thrown by the same blamed commit.  I'm writing this as separate issue since I'm using the S3Client, but I imagine they should be fixed together.

**Comment by ahmedelhaw** [comment]
"@aws-sdk/client-s3": "3.20.0",
 "@aws-sdk/credential-providers": "^3.40.0",

not sure if it's the upgrade for the latter or the downgrade of the former :)
works.

**Comment by ajredniwja** [comment]
I see the PR for the fix was merged, is it still a persisting issue?

**Comment by JJSphar** [comment]
Looking at the changelog, it seems the fix made it out in version 3.41.0.

Some quick testing on v3.44 shows that I can now perform S3 GetObjectRequests (using Credentials) in a production Webpack build.

Thanks!
